app/models/setting.rb

```diff
@@ -272,7 +272,12 @@ def self.load_defaults
     true
   end
 
-  def self.set(name, description, default, full_name = nil, value = nil)
+  def self.set(name, description, default, full_name = nil, value = nil, options = {})
+    if options.has_key? :collection
+      SettingsHelper.module_eval do
```

Comment:
Adds a dynamic method to `SettingsHelper`. 
the method returns the possible setting's values collection.